### ISSUE TITLE
feat: optimize() should route through forward() to support NU mesh

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#64]
## Problem

`optimize()` and `topology_optimize()` call `sim._forward_from_materials()` directly, bypassing the public `sim.forward()` path. This means:

1. Non-uniform mesh is not available for `optimize()` — NU support lives in `forward()` → `_forward_nonuniform_from_materials()`
2. New features added to `forward()` (checkpoint_every, emit_time_series) are not available in `optimize()`
3. The API split creates maintenance burden — features must be added in two places

## Current Code Path

```
optimize()  → _forward_from_materials()  → uniform runner only
forward()   → _forward_nonuniform_from_materials()  → NU + checkpoint + emit_ts
```

## Desired Code Path

```
optimize()  → forward()  → dispatches to uniform or NU automatically
```

## Impact

Users who want to combine `DesignRegion` continuous optimization with NU mesh (e.g., DRA on thin substrate with fine z-cells in substrate) currently cannot use the `optimize()` convenience API. They must write a manual `jax.value_and_grad` loop with `forward(eps_override=...)`.

Same for `topology_optimize()` — though soft PEC on NU is a separate blocker (#45).

## Suggested Approach

Refactor `optimize()` internal forward call from:
```python
result = sim._forward_from_materials(grid, materials, ...)
```
to:
```python
result = sim.forward(eps_override=eps_r, checkpoint=True, n_steps=_n_steps)
```

This requires `forward()` to handle the material assembly internally (which it already does for the NU path).

**Comment by bk-squared** [comment]
Implemented in `70a8be9`. `optimize()` now builds `eps_override` from the design region overlay and delegates to `sim.forward()` — uniform/NU dispatch happens inside `forward()`. New kwargs forwarded through: `checkpoint_every`, `n_warmup`, `emit_time_series`.

Test: `tests/test_optimize_nonuniform.py` — dz_profile optimize() now produces finite loss (previously raised ValueError). 27 existing optimize/auto-preflight tests still pass.
